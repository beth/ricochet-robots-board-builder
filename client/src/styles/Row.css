:local .row {
  display: grid;
  grid-template-columns: repeat(16, 50px);
}
