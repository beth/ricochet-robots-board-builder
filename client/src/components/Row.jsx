import React from 'react';
import PropTypes from 'prop-types';
import Square from './Square';
import styles from '../styles/Row.css';

const Row = props => (
  <div className={styles.row}>
    {props.squares.map(square => <Square square={square} key={square.col} />)}
  </div>
);

Row.propTypes = {
  squares: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Row;
