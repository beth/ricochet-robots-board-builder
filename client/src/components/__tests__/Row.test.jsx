import React from 'react';
import renderer from 'react-test-renderer';
import Row from '../Row';
import BoardConstructor from '../../lib/Board';

describe('Row Component', () => {
  it('renders correctly', () => {
    const board = new BoardConstructor(8);
    const domTree = renderer.create(<Row squares={board.squares[0]} />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
