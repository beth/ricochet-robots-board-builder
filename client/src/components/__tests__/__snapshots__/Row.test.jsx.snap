// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Component renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="top left square"
  />
  <div
    className="top square"
  />
  <div
    className="top square"
  />
  <div
    className="top square"
  />
  <div
    className="top square"
  />
  <div
    className="top square"
  />
  <div
    className="top square"
  />
  <div
    className="top square"
  />
  <div
    className="top square"
  />
  <div
    className="top square"
  />
  <div
    className="top square"
  />
  <div
    className="top square"
  />
  <div
    className="top square"
  />
  <div
    className="top square"
  />
  <div
    className="top square"
  />
  <div
    className="top right square"
  />
</div>
`;
